homeassistant/components/kmtronic/switch.py

```diff
@@ -28,22 +30,23 @@ async def async_setup_entry(
 class KMtronicSwitch(CoordinatorEntity, SwitchEntity):
     """KMtronic Switch Entity."""
 
-    def __init__(self, coordinator, relay, reverse, config_entry_id):
+    def __init__(self, hub, coordinator, relay, reverse, config_entry_id):
         """Pass coordinator to CoordinatorEntity."""
         super().__init__(coordinator)
         self._relay = relay
         self._config_entry_id = config_entry_id
         self._reverse = reverse
 
-    @property
-    def name(self) -> str:
-        """Return the name of the entity."""
-        return f"Relay{self._relay.id}"
+        hostname = urllib.parse.urlsplit(hub.host).hostname
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, self._config_entry_id)},
+            "name": f"Controller {hostname}",
+            "manufacturer": MANUFACTURER,
+            "configuration_url": hub.host,
+        }
 
-    @property
-    def unique_id(self) -> str:
-        """Return the unique ID of the entity."""
-        return f"{self._config_entry_id}_relay{self._relay.id}"
+        self._attr_name = f"Relay{self._relay.id}"
+        self._attr_unique_id = f"{self._config_entry_id}_relay{self._relay.id}"
```

Comment:
You can use the original arguments here.
```suggestion
        self._attr_name = f"Relay{relay.id}"
        self._attr_unique_id = f"{config_entry_id}_relay{relay.id}"
```